src/main/java/org/aksw/iguana/commons/io/BigByteArrayInputStream.java

```diff
@@ -1,54 +1,101 @@
 package org.aksw.iguana.commons.io;
 
+import org.apache.hadoop.hbase.io.ByteArrayOutputStream;
+
 import java.io.IOException;
 import java.io.InputStream;
 
+import static java.lang.Math.min;
+
 public class BigByteArrayInputStream extends InputStream {
```

Comment:
It should have something to mark it as closed/done so that we can be sure it was written until the (intended) end.

Reply:
Some more explanation: It makes the usage of the classes safer, if it is mad explicit that the input stream cannot be used until the output stream is done (e.g., until it has been closed) and that the output stream doesn't take data after it has been closed. This avoids that the streams are used in parallel in different threads.

So my suggestion (although it can be done in a different way as well):
- in all write methods, check whether the output stream is closed. If yes, throw an exception, else write data as usual.
- in all read methods, check whether the connected output stream is closed. If yes, read as usual, else throw an exception.